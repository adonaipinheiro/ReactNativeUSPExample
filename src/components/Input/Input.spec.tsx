import React from 'react';

import { fireEvent, render, act } from '@testing-library/react-native';

import { Input } from './';

describe('Input', () => {
    it('renders correctly with placeholder', () => {
        const { getByPlaceholderText, toJSON } = render(
            <Input placeholder="Type here" />
        );

        const input = getByPlaceholderText('Type here');
        expect(input).toBeTruthy();
        expect(toJSON()).toMatchSnapshot();
    });

    it('applies the style from styles.container', () => {
        const { getByPlaceholderText } = render(
            <Input placeholder="Test input" />
        );

        const input = getByPlaceholderText('Test input');
        expect(input.props.style).toBeDefined();
    });

    it('calls onChangeText when typing', () => {
        const handleChange = jest.fn();
        const { getByPlaceholderText } = render(
            <Input placeholder="Type" onChangeText={handleChange} />
        );

        const input = getByPlaceholderText('Type');
        fireEvent.changeText(input, 'Hello');
        expect(handleChange).toHaveBeenCalledWith('Hello');
    });

    it('should re-render if props change', () => {
        const { getByPlaceholderText, rerender } = render(
            <Input placeholder="First" />
        );

        act(() => {
            rerender(<Input placeholder="Second" />);
        });

        const input = getByPlaceholderText('Second');
        expect(input).toBeTruthy();
    });

    it('should not re-render if props do not change', () => {
        const renderSpy = jest.fn();

        const Wrapper = ({ placeholder }: { placeholder: string }) => {
            // o renderSpy() está contando as renderizações do Wrapper
            renderSpy();
            return <Input placeholder={placeholder} />;
        };

        const { rerender } = render(<Wrapper placeholder="Same" />);
        rerender(<Wrapper placeholder="Same" />);

        expect(renderSpy).toHaveBeenCalledTimes(2);
    });
});
