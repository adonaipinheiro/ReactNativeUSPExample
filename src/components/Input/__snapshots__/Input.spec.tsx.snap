// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly with placeholder 1`] = `
<TextInput
  placeholder="Type here"
  style={
    {
      "borderColor": "#0006",
      "borderRadius": 4,
      "borderWidth": 1,
      "fontSize": 16,
      "height": 40,
      "padding": 10,
      "width": "100%",
    }
  }
/>
`;
